src/Microsoft.AspNetCore.ResponseCaching/Interfaces/IKeyProvider.cs

```diff
@@ -8,18 +8,25 @@ namespace Microsoft.AspNetCore.ResponseCaching
     public interface IKeyProvider
     {
         /// <summary>
-        /// Create a key using the HTTP request.
+        /// Create a base key using the HTTP request.
         /// </summary>
         /// <param name="httpContext">The <see cref="HttpContext"/>.</param>
         /// <returns>The created base key.</returns>
         string CreateBaseKey(HttpContext httpContext);
 
         /// <summary>
-        /// Create a key using the HTTP context and vary rules.
+        /// Create a vary key using the HTTP context and vary rules.
         /// </summary>
         /// <param name="httpContext">The <see cref="HttpContext"/>.</param>
         /// <param name="varyRules">The <see cref="VaryRules"/>.</param>
-        /// <returns>The created base key.</returns>
+        /// <returns>The created vary key.</returns>
         string CreateVaryKey(HttpContext httpContext, VaryRules varyRules);
+
+        /// <summary>
+        /// Create a key for the response body using the HTTP context.
+        /// </summary>
+        /// <param name="httpContext">The <see cref="HttpContext"/>.</param>
+        /// <returns>The created body key.</returns>
+        string CreateBodyKey(HttpContext httpContext);
```

Comment:
Though this is sufficient for us, it doesn't really feel right. The logic should be: given an HttpContext and a CachedResponse without a body, create a body key. I'm also not keen on exposing our internal storage representation of a CachedResponse.